pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarMessage.java

```diff
@@ -1009,7 +1009,7 @@ public void setDoubleProperty(String name, double value) throws JMSException {
   public void setStringProperty(String name, String value) throws JMSException {
     checkWritableProperty(name);
     properties.put(name, value);
-    // not type, not needed
+    properties.put(propertyType(name), "string");
```

Comment:
In case you have tens of string properties the cost of this property becomes non negligible and you are going to spend much money for storage.

If you really need this  (and I understand that maybe you do) then I suggest to add a feature flag in the PulsarConnectionFactory

Reply:
an alternative is to ensure that when you set this to "string" you call 
` properties.remove(propertyType(name)`
and ensure that the old type is no more there